near-sdk/src/types/gas.rs

```diff
@@ -0,0 +1,140 @@
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use core::ops;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+/// Represents the amount of NEAR tokens in "gas units" which are used to fund transactions.
+#[derive(
+    Default,
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    PartialOrd,
+    Ord,
+    Eq,
+    BorshSerialize,
+    BorshDeserialize,
+    Hash,
+    BorshSchema,
+)]
+pub struct Gas(pub u64);
+
+impl Gas {
+    pub const fn new(amount: u64) -> Self {
+        Self(amount)
+    }
+}
+
+impl Serialize for Gas {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut buf = [0u8; 20];
```

Comment:
Let's add a test that we can round-trip `Gas(u64::max)`, to make sure the buffer size is enough.

Reply:
I already did, it's in this PR (checked with max, `0` and arbitrary `8`)

Reply:
ah, I looked at that test and somehow missed that :)